src/main/java/org/candlepin/auth/UserPrincipal.java

```diff
@@ -65,10 +66,37 @@ public UserPrincipal(String username, Collection<? extends Permission> permissio
         addPermission(new AsyncJobStatusPermission(getData(), this.getOwnerIds()));
     }
 
+    public UserPrincipal(String username, Collection<? extends Permission> permissions, boolean admin,
+                         OwnerInfo primaryOwner, AuthenticationMethod authMethod) {
+        this.username = username;
+        this.admin = admin;
+        this.primaryOwner = primaryOwner;
+        this.authMethod = authMethod == null ? AuthenticationMethod.BASIC : authMethod;
```

Comment:
I think if authMethod is null, it is better to leave it as null, since setting BASIC as default would be misleading.